### PR TITLE
PR: Add warning message about updating Spyder if installed with Anaconda

### LINKED ISSUE #4173: Add a message in the update dialog about not using pip if Spyder is running under Anaconda/Miniconda
@andfoy, the message should detect if Spyder is running under Anaconda or Miniconda and then it should warn users in our Update dialog like this:

> **Important note**: If you are using Spyder with Anaconda or Miniconda, please **don't** use `pip` to update it because that will (most probably) break your installation.<br>
Instead, please wait until new conda packages are made available and use `conda update spyder` to do it.

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #4173

### CODE REVIEW

File: spyder/app/mainwindow.py [hunk 1]
```diff
@@ -2776,13 +2776,25 @@ def _check_updates_ready(self):
             check_updates = box.is_checked()
         else:
             if update_available:
+                anaconda_msg = ''
+                if 'Anaconda' in sys.version or 'conda-forge' in sys.version:
+                    anaconda_msg = _("<b>Important note: </b> It seems that "
+                                     "you are using Spyder with Anaconda/"
+                                     "Miniconda, please <b>don't</b> use pip "
+                                     "to update as it would probably break "
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: Please change this line to

> to update it as that would probably break

File: spyder/app/mainwindow.py [hunk 2]
```diff
@@ -2776,13 +2776,25 @@ def _check_updates_ready(self):
             check_updates = box.is_checked()
         else:
             if update_available:
+                anaconda_msg = ''
+                if 'Anaconda' in sys.version or 'conda-forge' in sys.version:
+                    anaconda_msg = _("<b>Important note: </b> It seems that "
+                                     "you are using Spyder with Anaconda/"
+                                     "Miniconda, please <b>don't</b> use pip "
+                                     "to update as it would probably break "
+                                     "your installation.<br><br> Instead, "
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: Please move `Instead` to a new line to reflect that we are in a new paragraph.

File: spyder/app/mainwindow.py [hunk 3]
```diff
@@ -2776,13 +2776,25 @@ def _check_updates_ready(self):
...
             if update_available:
+                anaconda_msg = ''
+                if 'Anaconda' in sys.version or 'conda-forge' in sys.version:
+                    anaconda_msg = _("<b>Important note: </b> It seems that "
+                                     "you are using Spyder with Anaconda/"
+                                     "Miniconda, please <b>don't</b> use pip "
+                                     "to update as it would probably break "
+                                     "your installation.<br><br> Instead, "
+                                     "please wait until new conda packages "
+                                     "are made available and use <code>conda "
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: Please change the text from this point to the end to

> are made available and use <b>conda</b> to perform the update.

The thing is people could have used conda-forge to install Spyder :-)

**Comment by goanpeca** [comment]
Would it help to have a separate mini QtApp (Or even TKinter app?) to handle updates of Spyder?

We could check for conda or miniconda and try to run conda or pip in a separate app to guarantee that updates work in a better way?

Thoughts? We are planning on doing this for navigator to have a more stable update process.

**Comment by andfoy** [comment]
@goanpeca Like a separate application named Spyder Update Manager or some similar utility?

**Comment by goanpeca** [comment]
>@goanpeca Like a separate application named Spyder Update Manager or some similar utility?

Exactly (but could be part of the spyder package by default) not a separate conda/pip package

File: spyder/app/mainwindow.py [hunk 4]
```diff
@@ -2776,13 +2776,24 @@ def _check_updates_ready(self):
             check_updates = box.is_checked()
         else:
             if update_available:
+                anaconda_msg = ''
+                if 'Anaconda' in sys.version or 'conda-forge' in sys.version:
+                    anaconda_msg = _("<b>Important note: </b> It seems that "
+                                     "you are using Spyder with Anaconda/"
+                                     "Miniconda, please <b>don't</b> use pip "
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: pip -> <tt>pip</tt>

File: spyder/app/mainwindow.py [hunk 5]
```diff
@@ -2776,13 +2776,24 @@ def _check_updates_ready(self):
             check_updates = box.is_checked()
         else:
             if update_available:
+                anaconda_msg = ''
+                if 'Anaconda' in sys.version or 'conda-forge' in sys.version:
+                    anaconda_msg = _("<b>Important note: </b> It seems that "
+                                     "you are using Spyder with Anaconda/"
+                                     "Miniconda, please <b>don't</b> use pip "
+                                     "to update as that would probably break "
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: to update **it** ...

File: spyder/app/mainwindow.py [hunk 6]
```diff
@@ -2776,13 +2776,24 @@ def _check_updates_ready(self):
...
         else:
             if update_available:
+                anaconda_msg = ''
+                if 'Anaconda' in sys.version or 'conda-forge' in sys.version:
+                    anaconda_msg = _("<b>Important note: </b> It seems that "
+                                     "you are using Spyder with Anaconda/"
+                                     "Miniconda, please <b>don't</b> use pip "
+                                     "to update as that would probably break "
+                                     "your installation.<br><br> "
+                                     "Instead, please wait until new conda "
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: conda -> <tt>conda</tt>

File: spyder/app/mainwindow.py [hunk 7]
```diff
@@ -2776,13 +2776,24 @@ def _check_updates_ready(self):
...
+                                     "packages are made available and use "
+                                     "<br>conda</br> to perform the update.")
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: <tt>conda</tt>

**Pull request merged**